tests/test_assembly.py

```diff
@@ -464,5 +465,67 @@ def threaded_nonsym(u, v, w):
         )
 
 
+@pytest.mark.parametrize(
+    "m,mdgtype,etype",
+    [
+        (
+            MeshTri.init_tensor(np.linspace(0, 1, 7),
+                                np.linspace(0, 1, 7)),
+            MeshTri1DG,
+            ElementTriP1,
+        ),
+        (
+            MeshTri.init_tensor(np.linspace(0, 1, 5),
+                                np.linspace(0, 1, 5)),
+            MeshTri1DG,
+            ElementTriP1,
+        ),
+        (
+            MeshTri().refined(2),
+            MeshTri1DG,
+            ElementTriP1,
+        ),
+        (
+            MeshTri().refined(3),
+            MeshTri1DG,
+            ElementTriP1,
+        ),
+        (
+            MeshQuad().refined(2),
+            MeshQuad1DG,
+            ElementQuad1,
+        ),
+        (
+            MeshQuad().refined(2),
+            MeshQuad1DG,
+            ElementQuad2,
+        ),
+        (
+            MeshTri().refined(2),
+            MeshTri1DG,
+            ElementTriP2,
+        ),
+    ]
+)
+def test_periodic_mesh_assembly(m, mdgtype, etype):
+    m = m.reordered(m.nodes_satisfying(lambda x: x[0] == 1))
+    m = m.with_boundaries({
+        'left': lambda x: x[0] == 0,
+        'right': lambda x: x[0] == 1,
+    })
+    mp = mdgtype.from_mesh(m, m.periodic_connectivity('right', 'left'))
```

Comment:
I want to somehow combine these three steps into one, but don't know yet what is the correct approach.